src/notelistmodel.cpp

```diff
@@ -453,7 +451,7 @@ bool NoteListModel::dropMimeData(const QMimeData *mime, Qt::DropAction action, i
                 auto lastMod = index.data(NoteDeletionDateTime).toDateTime();
                 for (destinationChild = 0; destinationChild < m_noteList.size();
                      ++destinationChild) {
-                    const auto &note = m_noteList[destinationChild];
+                    note = m_noteList[destinationChild];
```

Comment:
Erm, this massively changes the behavior. Previously, it was storing a read-only `NodeData` reference to `m_noteList[destinationChild]`, which was used to inspect the `deletionDateTime`. Whereas the new line copies `m_noteList[destinationChild]` into the previously referenced node data instead.

(also applies below)

Reply:
You're absolutely right, thank you for spotting this!